### PR TITLE
fix: type check records correctly

### PR DESCRIPTION
**Pull request by tyleroooo**
Now the records are correctly checking for exhaustivity and value types matching.

### CODE REVIEW

File: src/components/Table.tsx [hunk 1]
```diff
@@ -76,7 +76,8 @@ export type ColumnDef<TableRowData extends BaseTableRowData | CustomRowConfig> =
   tag?: React.ReactNode;
   colspan?: number;
   childColumns?: ColumnDef<TableRowData>[];
-  getCellValue: (row: TableRowData) => string | number;
+  // if sorting is false, can pass undefined for getCellValue
+  getCellValue: ((row: TableRowData) => string | number | undefined | null) | undefined;
```
Comment (moo-onthelawn) [inline src/components/Table.tsx:80]: hmm to enforce this in typescript would it be possible to do something like 

```
type SortableColumnDef = {
  allowsSorting: true;
  getCellValue: ((row: TableRowData) => string | number | undefined | null);
}

type NonSortableColumnDef = {
  allowsSorting: false;
}

type ColumnDef<...>  = {
...
} & (SortableColumnDef || NonSortableColumnDef)
```

alternatively, is this just equivalent to `getCellValue?: ((row: TableRowData) => string | number | undefined | null)`

File: src/views/tables/OrdersTable.tsx [hunk 2]
```diff
@@ -210,6 +210,7 @@ const getOrdersTableColumnDef = ({
         label: stringGetter({ key: STRING_KEYS.ACTION }),
         isActionable: true,
         allowsSorting: false,
+        getCellValue: undefined,
```
Comment (moo-onthelawn) [inline src/views/tables/OrdersTable.tsx:213]: I think if the above typing works, we'd maybe be able to just not pass in an argument here

File: src/components/Table.tsx (hunk 1 above)
Comment (moo-onthelawn) [reply src/components/Table.tsx:80]: another thought, is it necessary for us to capture `undefined` and `null` in the return types? (I sort of prefer just using one or the other, `null` seems to make more sense in this case)

**Review by moo-onthelawn — APPROVED** [review]
lgtm with typing consideration

File: src/components/Table.tsx (hunk 1 above)
Comment (tyleroooo) [reply src/components/Table.tsx]: YEES great idea! That's much cleaner.

**Pull request merged**